enhancements/scheduling/prevent-user-workloads-from-being-scheduled-on-control-plane-nodes.md

```diff
@@ -207,12 +221,18 @@ spec:
       operations:  ["CREATE", "UPDATE"]
       resources:   ["pods"]
   validations:
-    - expression: "object.spec.tolerations.all(toleration, !(toleration.key == 'node-role.kubernetes.io/control-plane' && toleration.effect == 'NoExecute'))"
+    - expression: "object.spec.tolerations.all(toleration, (
+      toleration.effect != 'NoExecute' ||
+      (toleration.effect == 'NoExecute' &&
+      (auhorizer.serviceAccount(object.metadata.namespace, object.spec.serviceAccountName).group('').resource(toleration.key).namespace(object.metadata.namespace).check(toleration.effect).allowed()))))"
```

Comment:
if `toleration.key` does not exist under the role's list (e.g. `default/tolerate-noexecute-role`) of resources, will the expression gets evaluated as true?

Reply:
You can test that. It won't. 

If the effect is not NoExecute it's allowed 

If the effect is NoExecute, it checks the authorizer, then it's basically normal rbac logic. Wrong keys, inexistent keys, or wrong verbs won't be allowed